lib/mvi/responses/profile_parser.rb

```diff
@@ -30,40 +20,76 @@ class FindCandidate < Base
       NAME_XPATH = 'patientPerson/name'
       ADDRESS_XPATH = 'patientPerson/addr'
       PHONE = 'patientPerson/telecom'
-      ACKNOWLEDGEMENT_DETAIL_XPATH = 'acknowledgement/acknowledgementDetail/text'
 
+      ACKNOWLEDGEMENT_DETAIL_XPATH = 'acknowledgement/acknowledgementDetail/text'
       MULTIPLE_MATCHES_FOUND = 'Multiple Matches Found'
 
+      # MVI response code options.
+      EXTERNAL_RESPONSE_CODES = {
+        success: 'AA',
+        failure: 'AE',
+        invalid_request: 'AR'
+      }.freeze
+
+      # Creates a new parser instance.
+      #
+      # @param response [struct Faraday::Env] the Faraday response
+      # @return [ProfileParser] an instance of this class
       def initialize(response)
-        super(response)
-        @subject = locate_element(@original_body, SUBJECT_XPATH)
+        @original_body = locate_element(response.body, BODY_XPATH)
+        @code = locate_element(@original_body, CODE_XPATH)
+      end
+
+      # MVI returns failed or invalid codes if the request is malformed or MVI throws an internal error.
+      #
+      # @return [Boolean] has failed or invalid code?
+      def failed_or_invalid?
+        result = [EXTERNAL_RESPONSE_CODES[:failure], EXTERNAL_RESPONSE_CODES[:invalid_request]].include? @code
+        Rails.logger.warn "MVI returned response with code: #{@code}" if result
```

Comment:
This looks like it might be interesting enough to log to Sentry as a warning so that we can see how often it occurs (see `sentry_logging.rb`).  Feel free to not do this if you already have a good handle on this info and don't want to pollute our Sentry logs.